OpenTelemetry.Azure.Sampling/ThirdPartyParentOnlySampler.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Azure.Sampling
+{
+    [Flags]
+    public enum PublicBoundary
+    {
+        Incoming = 1,
+        Outgoing = 2,
+    }
+
+    /// <summary>
+    /// Sampler implements parent-only strategy: samples in spans following client decision
+    /// </summary>
+    public class ThirdPartyParentOnlySampler : Sampler
+    {
+        private const string UserFlagName = "tf@az";
+        private static readonly int UserFlagLength = "tf@az".Length;
+
+        private readonly Func<string> getResourceIdCallback;
+        private readonly Func<string, bool> samplingCallback;
+        private readonly PublicBoundary boundaries;
+        private readonly bool propagateContext;
+
+        public ThirdPartyParentOnlySampler(Func<string, bool> samplingCallback, Func<string> getResourceIdCallback, PublicBoundary boundaries, bool propagateContext = true)
+        {
+            this.getResourceIdCallback = getResourceIdCallback;
+
+            // sampling callback will be different for rate-based sampling: Func<string, double, bool> samplingCallback
+
+            this.samplingCallback = samplingCallback;
+            this.boundaries = boundaries;
+            this.propagateContext = propagateContext;
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            // if it's server span, and it's on the public endpoint
+            if ((this.boundaries & PublicBoundary.Incoming) == PublicBoundary.Incoming &&
+                this.IsServerSpan(in samplingParameters))
+            {
+                string resourceId = this.getResourceIdCallback?.Invoke();
+
+                // follow parent decision + respect sampling configuration settings
+                if (!string.IsNullOrEmpty(resourceId) && 
+                    samplingParameters.ParentContext.TraceFlags == ActivityTraceFlags.Recorded &&
+                    this.samplingCallback(resourceId))             // samplingCallback(resoureId, GetScore(traceId)) <= this.rate for rate-based
```

Comment:
So for rate based is the idea to have a GetScore method that would calculate the score and then the callback could make a decision.

Reply:
yes, getscore will be an internal rate-based sampler implementation detail